src/pages/Search/SearchFiltersTaxRatePage.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {useCallback, useMemo} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SearchMultipleSelectionPicker from '@components/SearchMultipleSelectionPicker';
+import type {SearchMultipleSelectionPickerItem} from '@components/SearchMultipleSelectionPicker';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import {getAllTaxRates} from '@libs/PolicyUtils';
+import * as SearchActions from '@userActions/Search';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+
+function SearchFiltersTaxRatePage() {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+    const allTaxRates = getAllTaxRates();
+    const selectedTaxesItems = searchAdvancedFiltersForm?.taxRate?.map((taxRate) => ({name: allTaxRates[taxRate].name, value: taxRate}));
+    const policyID = searchAdvancedFiltersForm?.policyID ?? '-1';
+    const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`);
+    const singlePolicyTaxRates = policy?.taxRates?.taxes;
+
+    const taxItems = useMemo(() => {
+        let taxRates: SearchMultipleSelectionPickerItem[] = [];
+        if (!singlePolicyTaxRates) {
+            taxRates = Object.entries(allTaxRates).map(([taxRateKey, taxRateInfo]) => ({name: taxRateInfo.name, value: taxRateKey}));
+        } else {
+            taxRates = Object.entries(singlePolicyTaxRates).map(([taxRateKey, taxRateInfo]) => ({name: taxRateInfo.name, value: taxRateKey}));
+        }
```

Comment:
I think we should display the tax name in the filter and use the ID for the API call. We do similar things for the card filter for example

Reply:
I think that's a good idea, at first I opposed this one but when I thought it through.
In case when we have two workspaces and we add tax rate with the same name to both workspaces(they might have different values), The tax rate key will be the same 😄 . So basically it's the same as we would search over tax name.
These are two different workspaces with the tax rate with the same name:
<img width="663" alt="Screenshot 2024-08-08 at 11 36 55" src="https://github.com/user-attachments/assets/cc9455f4-62e5-4478-ae92-88169c54024d">
<img width="564" alt="Screenshot 2024-08-08 at 11 36 15" src="https://github.com/user-attachments/assets/6d4db683-d9fa-4b7a-92e5-b62a3dc03b43">

Reply:
WDYT @luacmartins

Reply:
Looking into the backend part, but that makes sense to me

Reply:
I took a look at the backend and unfortunately we have no way to filter tax rates without the taxRateID, so we must send that. I think we could eliminate duplicated names like this (in a follow up PR):

1. Display de-duped names only
2. If more than one taxRateID for a given tax name exists, send both taxRateIDs to the API
3. The API will filter them by `WHERE taxRateID IN (<ids>)`

Reply:
I've thought this through and I think it's fine.
2. If more than one taxRateID for a given tax name exists, send both IDs to the API
this situation won't be possible because externalID is created based on taxRate name. So I just need to send one externalID.
Case closed I'll de-dup names and everything should work fine!

Reply:
I don't think we update the externalID if you rename the tax though, do we?